### PR TITLE
Allow custom helm values

### PR DESCRIPTION
**Pull request by atanasdinov**
- Adds support for additional Helm values in the ReleaseManifest custom resource
- Adds support for additional Helm values in the UpgradePlan custom resource
- User supplied values (via UpgradePlan spec) have higher priority than platform supplied values (via ReleaseManifest spec)

### CODE REVIEW

File: internal/controller/helm.go [hunk 1]
```diff
@@ -125,6 +144,75 @@ func (r *UpgradePlanReconciler) createHelmChart(ctx context.Context, upgradePlan
...
+}
+
+func mergeMaps(m1, m2 map[string]any) map[string]any {
+	out := make(map[string]any, len(m1))
+	for k, v := range m1 {
+		out[k] = v
+	}
+
+	for k, v := range m2 {
+		if v, ok := v.(map[string]any); ok {
```
Comment (ipetrov117) [inline internal/controller/helm.go:204]: Nit: naming is a bit hard to follow, can we have something like:
```suggestion
		if inner, ok := v.(map[string]any); ok {
```

File: internal/controller/helm.go [hunk 2]
```diff
@@ -125,6 +144,75 @@ func (r *UpgradePlanReconciler) createHelmChart(ctx context.Context, upgradePlan
...
+			if inner, ok := out[k].(map[string]any); ok {
```
Comment (ipetrov117) [inline internal/controller/helm.go:205]: Nit:
```suggestion
			if outInner, ok := out[k].(map[string]any); ok {
```

File: internal/controller/helm_test.go [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+				"a": 1,
+			},
+			m2: map[string]any{},
+			result: map[string]any{
+				"a": 1,
+			},
+		},
+		{
+			name: "empty first map, non-empty second map",
+			m1:   map[string]any{},
```
Comment (ipetrov117) [inline internal/controller/helm_test.go:37]: Nit: indentation
```suggestion
			m1: map[string]any{},
```

**Review by ipetrov117 — DISMISSED** [review]
Overall the code looks good. We need to pay special attention to this, when we do our testing.

File: internal/controller/helm_test.go (hunk 3 above)
Comment (atanasdinov) [reply internal/controller/helm_test.go:37]: Looks ugly, indeed, but this is the default formatting for empty map under the string above - it'd be as suggested if it was populated.

Comment (ipetrov117) [reply internal/controller/helm_test.go:37]: oki, my bad.

**Review by ipetrov117 — APPROVED** [review]
